Sources/PubNub/EventEngine/Subscribe/Helpers/SubscribeRequest.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  SubscribeRequest.swift
+//
+//  Copyright (c) PubNub Inc.
+//  All rights reserved.
+//
+//  This source code is licensed under the license found in the
+//  LICENSE file in the root directory of this source tree.
+//
+
+import Foundation
+
+class SubscribeRequest {
+  let channels: [String]
+  let groups: [String]
+  let timetoken: Timetoken?
+  let region: Int?
+  
+  private let configuration: SubscriptionConfiguration
+  private let session: SessionReplaceable
+  private let sessionResponseQueue: DispatchQueue
+  private let channelStates: [String: [String: JSONCodableScalar]]
+  
+  private var request: RequestReplaceable?
+    
+  var retryLimit: UInt { configuration.automaticRetry?.retryLimit ?? 0 }
+  var onAuthChallengeReceived: (() -> Void)?
+  
+  init(
+    configuration: SubscriptionConfiguration,
+    channels: [String],
+    groups: [String],
+    channelStates: [String: [String: JSONCodableScalar]],
+    timetoken: Timetoken? = nil,
+    region: Int? = nil,
+    session: SessionReplaceable,
+    sessionResponseQueue: DispatchQueue
+  ) {
+    self.configuration = configuration
+    self.channels = channels
+    self.groups = groups
+    self.channelStates = channelStates
+    self.timetoken = timetoken
+    self.region = region
+    self.session = session
+    self.sessionResponseQueue = sessionResponseQueue
+    
```

Comment:
Should `state` also be somewhere here?

Reply:
It's already present [here](https://github.com/pubnub/swift/blob/feat/ee/Sources/PubNub/EventEngine/Subscribe/Helpers/SubscribeRequest.swift#L33)